File: internal/utils/response.go
package utils

import (
	"net/http"

	"github.com/gofiber/fiber/v2"
)

type Response struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

type PaginationResponse struct {
	Total    int64       `json:"total"`
	Page     int         `json:"page"`
	PageSize int         `json:"page_size"`
	List     interface{} `json:"list"`
}

func Success(c *fiber.Ctx, data interface{}) error {
	return c.JSON(Response{
		Code:    0,
		Message: "success",
		Data:    data,
	})
}

func SuccessWithPagination(c *fiber.Ctx, list interface{}, total int64, page, pageSize int) error {
	return c.JSON(Response{
		Code:    0,
		Message: "success",
		Data: PaginationResponse{
			Total:    total,
			Page:     page,
			PageSize: pageSize,
			List:     list,
		},
	})
}

func Error(c *fiber.Ctx, code int, message string) error {
	return c.Status(http.StatusBadRequest).JSON(Response{
		Code:    code,
		Message: message,
	})
}

func ErrorWithStatus(c *fiber.Ctx, statusCode, code int, message string) error {
	return c.Status(statusCode).JSON(Response{
		Code:    code,
		Message: message,
	})
}

func Unauthorized(c *fiber.Ctx, message string) error {
	return ErrorWithStatus(c, http.StatusUnauthorized, 401, message)
}

func Forbidden(c *fiber.Ctx, message string) error {
	return ErrorWithStatus(c, http.StatusForbidden, 403, message)
}

func NotFound(c *fiber.Ctx, message string) error {
	return ErrorWithStatus(c, http.StatusNotFound, 404, message)
}

func ValidationError(c *fiber.Ctx, message string) error {
	return ErrorWithStatus(c, http.StatusUnprocessableEntity, 422, message)
}

File: src/utils/response.ts
import { Response } from 'express';
import { ApiResponse, PaginatedResponse } from '../types';

export function success<T>(res: Response, data: T, message: string = '操作成功') {
  const response: ApiResponse<T> = {
    success: true,
    data,
    message,
  };
  return res.status(200).json(response);
}

export function paginated<T>(
  res: Response,
  data: T[],
  total: number,
  page: number,
  pageSize: number,
  message: string = '查询成功'
) {
  const response: ApiResponse<PaginatedResponse<T>> = {
    success: true,
    data: {
      data,
      total,
      page,
      pageSize,
      totalPages: Math.ceil(total / pageSize),
    },
    message,
  };
  return res.status(200).json(response);
}

export function error(res: Response, message: string, statusCode: number = 400, error?: string) {
  const response: ApiResponse = {
    success: false,
    message,
    error,
  };
  return res.status(statusCode).json(response);
}

export function unauthorized(res: Response, message: string = '未授权访问') {
  return error(res, message, 401);
}

export function forbidden(res: Response, message: string = '权限不足') {
  return error(res, message, 403);
}

export function notFound(res: Response, message: string = '资源不存在') {
  return error(res, message, 404);
}

export function serverError(res: Response, err: any) {
  console.error('Server Error:', err);
  return error(res, '服务器内部错误', 500, err.message);
}
